stripe.go

```diff
@@ -611,6 +613,14 @@ func (s *BackendImplementation) UnmarshalJSONVerbose(statusCode int, body []byte
 	return nil
 }
 
+// Regular expressions used to match a few error types that we know we don't
+// want to retry. Unfortunately these errors aren't typed so we match on the
+// error's message.
+var (
+	redirectsErrorRE = regexp.MustCompile(`stopped after \d+ redirects\z`)
+	schemeErrorRE    = regexp.MustCompile(`unsupported protocol scheme`)
```

Comment:
What fires those errors exactly? I'm not sure we should ever attempt something based on an error message

Reply:
They're fired from inside the `net/http` internals. String matching is definitely not great, but unless Go Core's blessed us with actual types, it's unfortunately the best we can do.

Reply:
But would it be the same message as Stripe's own error message or are those entirely separate?

Reply:
Entirely separate — they're hard-coded from pretty deep inside the Go standard libraries. For example:

https://github.com/golang/go/blob/d140c357442471ca0d56615811fb4226e99fadd5/src/net/http/transport.go#L538